src/include/messenger/messenger.h

```diff
@@ -119,31 +123,25 @@ class Messenger : public common::DedicatedThreadTask {
    * @param connection_id   The connection to send the message over.
    * @param message         The message to be sent.
    * @param callback        The callback function to be invoked on the response.
+   * @param recv_msg_id     The receiver's message ID, e.g., sent in response or to invoke preregistered functions.
+   *                        To invoke preregistered functions, use static_cast<uint8_t>(Messenger::BuiltinCallback).
```

Comment:
Why is the callback enum number being used as a recv_msg_id?

Reply:
I guess a more direct naming for this id from its usage is: `callback_id`, or `recv_cb_id`

Reply:
Renamed. Perhaps we should just call everything callback ID since that is what it really is..